src/node_jsvmapi.cc

```diff
@@ -481,13 +481,11 @@ namespace v8impl {
   RETURN_STATUS_IF_FALSE(!((maybe).IsNothing()), (status))
 
 #define NAPI_PREAMBLE(e)                                                \
-  do {                                                                  \
-    CHECK_ARG(e);                                                       \
-    RETURN_STATUS_IF_FALSE(v8impl::TryCatch::lastException().IsEmpty(), \
-      napi_pending_exception);                                          \
-    napi_clear_last_error();                                            \
-    v8impl::TryCatch tryCatch(v8impl::V8IsolateFromJsEnv((e)));         \
-  } while(0)
+  CHECK_ARG(e);                                                         \
+  RETURN_STATUS_IF_FALSE(v8impl::TryCatch::lastException().IsEmpty(),   \
+    napi_pending_exception);                                            \
+  napi_clear_last_error();                                              \
+  v8impl::TryCatch tryCatch(v8impl::V8IsolateFromJsEnv((e)))
```

Comment:
The do-while had to be removed because the `tryCatch` must have function scope. I made the same fix in my PR https://github.com/nodejs/abi-stable-node/pull/72 after the problem was revealed by test failures.

The other macros may continue to use do-while because it makes them behave more like a single statement rather than a statement group; as you're probably aware that's a common pattern. It's not strictly necessary just for our internal use though.